### DISCUSSION TITLE
Running a function (e.g. comment-region) on inserted region in `:post`

### DISCUSSION THREAD

**Discussion by ArsenArsen** [minad/tempel#131] [Q&A] [answered]
hi!

I'm trying to implement templates that universally (wrt mode) emits a copyright comment.  My goal is something akin to the following:

```lisp
(copyright
 (p "Brief file description") n
 "Copyright (C) " copyright-current-year
 "  " user-full-name " <" user-mail-address ">" n
 ; etc
 :post (comment-region (mark t) (point)))
```

... but, I don't see how to get the inserted region in the context of `:post` (namely, the mark appears to not be set often).  is this possible?  is it an XYproblem?

TIA, have a lovely day.

PS: I've considered the alternative of comment-{start,continue,end}, but the logic for commenting regions is not quite as simple as just inserting those values in the right spot in a comment, so I wanted to reuse the newcomment.el logic

**Comment by pmiam** [comment, marked as answer]
hello!

The mark needs to be set explicitly. This is usually done interactively by the user during the normal navigation of text, but can also be done while the template is being "auto-typed" by evaluating s-exps at strategic points. For instance:

```emacs-lisp
(cpr (push-mark) "Copyright (C) " copyright-current-year "  "
     user-full-name " <" user-mail-address ">" n
     :post (comment-region (mark) (point)))
```

This should work well as a standalone template, but I did find that it causes problems when included in a larger template using the `tempel-include` element as defined in the README. Trying to use this form in that way causes the encapsulating template to fail before evaluating `comment-region` with the error message `Lisp nesting exceeds 'max-lisp-eval-depth': 1601`

I've not been able to understand why that happens yet.

**Reply by ArsenArsen** [reply]
that was a good idea!  I've managed to make it work via:

```lisp

(gpl3+
 (require 'copyright)
 (progn (setq arsen--template-start-mark (point)) nil)
 (p "Brief file description") n
 "Copyright (C) " copyright-current-year
 "  " user-full-name " <" user-mail-address ">" n

 n
 "This program is free software: you can redistribute it and/or modify" n
 "it under the terms of the GNU General Public License as published by" n
 "the Free Software Foundation, either version 3 of the License, or" n
 "(at your option) any later version." n
 n
 "This program is distributed in the hope that it will be useful," n
 "but WITHOUT ANY WARRANTY; without even the implied warranty of" n
 "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the" n
 "GNU General Public License for more details." n
 n
 "You should have received a copy of the GNU General Public License" n
 "along with this program.  If not, see <http://www.gnu.org/licenses/>." n
 q
 :post (comment-region arsen--template-start-mark (point)))
```

I haven't tested your include issue, I'll try to "abstract" this later
